src/notebook.cpp

```diff
@@ -518,3 +519,13 @@ QList<QByteArray> Notebook::customPropertyKeys() const
 {
     return d->mCustomProperties.uniqueKeys();
 }
+
+QString Notebook::emailAddress() const
+{
+    if (!d->mPluginName.isEmpty()) {
+        Ptr ptr(new Notebook(*this));
```

Comment:
The notebook is more important detail than being a pointer or not, could use name more towards that.